### DISCUSSION TITLE
Li ion cylindrical model with sub-models.

### DISCUSSION THREAD

**Discussion by balaji3112** [pybamm-team/PyBaMM#2761] [Q&A] [answered]
I'm attempting to create a Li-ion cylindrical cell model in Pybamm, as well as integrate that model with a thermal system and a cooling system that uses phase change material. Are there any resources or references available to get started?

**Comment by RuiheLi** [comment, marked as answer]
Hi~ If you are a new starter on PyBaMM, maybe worth playing around some cases in the example folder [here](https://github.com/pybamm-team/PyBaMM/tree/develop/examples). I am not sure whether your thermal system or cooling system require the model to be 3D. As far as I know, PyBaMM is mainly for P2D level on electrochemical model, but can do 2D current collector, like [this](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/pouch-cell-model.ipynb). This [paper ](https://iopscience.iop.org/article/10.1149/1945-7111/aba44b/pdf) seems to have done something you may be interested, but it is PyBaMM coupled with OpenPNM.

**Comment by balaji3112** [comment]
Thanks for the resources.
